operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -203,6 +207,15 @@ private String getEffectiveLogHome() {
     return logHome;
   }
 
+  private String getEffectiveDataHome() {
+    String dataHome = getDataHome();
+    if (dataHome == null) {
+      // dataHome not specified, use default value
+      return DEFAULT_DATA_HOME + File.separator + getDomainUID();
+    }
+    return dataHome;
```

Comment:
If not specified, the value should be ignored. IMO this is necessary to (a) ensure backward compatibility of old operator domain configuration, (b) provide an intuitive way for users to tell us to use the configured value in the config.xml, or the default mbean value, instead of the override value, and most of all (c) prevent dataloss on upgrade as otherwise the restarted stores will not be able to find their original data.

Reply:
Confirmed done/resolved.